src/models/NavItemPage.php

```diff
@@ -205,12 +205,13 @@ private function renderPlaceholderRecursive($navItemPageId, $placeholderVar, $pr
             $prev = $key-1;
             $next = $key+1;
             $cacheKey = ['blockcache', (int) $placeholder['id']];
-            
+            $cacheKeyAssets = array_merge($cacheKey, ['assetsBundels']);
+
             $blockResponse = $this->getHasCache($cacheKey);
-            
+
             if ($blockResponse === false) {
                 
-                /** @var $blockObject \luya\cms\base\InternalBaseBlock */
+                /** @var $blockObject \luya\cms\base\PhpBlock */
```

Comment:
Actually its InternalBaseBlock, but since we only support PhpBlock for the moment we can do this. I will fix this in future. But maybe it should be a function inside EveryInternalBaseBLock (which is abstracted) where the logic should happen, cause you can also create your own base blocks. So maybe introducing a `onRegister()` method in block interface which will be called.

Reply:
What is you intention with `onRegister`? Where this method should be called?

Reply:
What is you intention with onRegister? Where this method should be called?

Reply:
I understand what you mean and I agree. But then we need two functions one for set the assets to the cache and a other to register from cache to the app view.